strategies/indicators/HEIKEN.js

```diff
@@ -0,0 +1,61 @@
+/*
+
+  Heiken-Ashi - manuel.wassermann97@gmail.com 21/01/2018
+
+ */
+
+
+var Indicator = function(){
+  this.input = 'candle';
+  this.high = null;
+  this.low = null;
+  this.open = null;
+  this.close = null;
+}
+
+
+
+Indicator.prototype.update = function(candle) {
+
+  // Only calculate if we have an previous candle
+  if(this.lastCandle){
+    
+    // heiken.close
+    this.xClose = function(previous, current){
+      return (previous.close+current.open+current.low+current.high)/4;
+    }  
+    
+    // first heiken.open 
+    // This is probably wrong
+    first = true;
+    if(first){
+      this.xOpen = function(previous){
+        return (previous.open+previous.close)/2;
+        first = false;
+      }
+
+    // heiken.open 
+    } else{
+      this.xOpen = function(previous){
+        return (this.xOpen(this.previousCandle)+this.xXlose(this.previousCandle))/2;
```

Comment:
xXlose, probably a type? You mean xClose I suspect.

Reply:
Thanks